### PR TITLE
I32 mixingmatrix

### LINKED ISSUE #32: inconsistent output classes for mixingmatrix
```
d> class(mixingmatrix(fit_40$network, "hiv")) # a network object
[1] "mixingmatrix"

d> class(mixingmatrix(main, "hiv")) # an egodata object
[1] "matrix" "array"
```

and the structure
```
str(networkmixmat)
List of 2
 $ type  : chr "undirected"
 $ matrix: 'table' num [1:2, 1:2] 1550 258 258 16
  ..- attr(*, "dimnames")=List of 2
  .. ..$ From: chr [1:2] "0" "1"
  .. ..$ To  : chr [1:2] "0" "1"
 - attr(*, "class")= chr "mixingmatrix"

str(egodatamixmat)
num [1:2, 1:2] 212.8 12.4 24.7 12.7
 - attr(*, "dimnames")=List of 2
  ..$ ego  : chr [1:2] "0" "1"
  ..$ alter: chr [1:2] "0" "1"
```

not sure what the rationale is for these different structures and properties.

assigning @krivit just to get his take.

### PR DESCRIPTION
**Pull request by mbojan**
Simpler object returned by `mixiningmatrix()`, closes #32:

- it is of class `c("mixingmatrix", "table")` with logical attributes `directed` and `bipartite`
- added `[[` and `$` methods for backward compatibility. They are `.Deprecated`
- made `is.directed()` and `is.bipartite()` generic with added methods for mixing matrices
- documented the value returned
- updated tests

### CODE REVIEW

**Comment by mbojan** [comment]
I will still fix via this PR:

```
── R CMD check results ───────────────────────────────── network 1.17.0-585 ────
Duration: 1m 5.5s

❯ checking Rd \usage sections ... WARNING
  Undocumented arguments in documentation object 'network.indicators'
    ‘...’
  
  Functions with \usage entries need to have the appropriate \alias
  entries, and all their arguments documented.
  The \usage entries must correspond to syntactically valid R code.
  See chapter ‘Writing R documentation files’ in the ‘Writing R
  Extensions’ manual.

❯ checking files in ‘vignettes’ ... WARNING
  Files in the 'vignettes' directory but no files in 'inst/doc':
    ‘networkVignette.Rnw’, ‘networkVignette.pdf’

❯ checking package vignettes in ‘inst/doc’ ... WARNING
  dir.exists(dir) is not TRUE
  Package vignette without corresponding single PDF/HTML:
     ‘networkVignette.Rnw’

❯ checking CRAN incoming feasibility ... NOTE
  Maintainer: ‘Carter T. Butts <buttsc@uci.edu>’
  
  Unknown, possibly mis-spelled, fields in DESCRIPTION:
    ‘Remotes’
  
  The Date field is over a month old.

❯ checking R code for possible problems ... NOTE
  print.mixingmatrix: no visible global function definition for
    ‘addmargins’
  Undefined global functions or variables:
    addmargins
  Consider adding
    importFrom("stats", "addmargins")
  to your NAMESPACE file.

0 errors ✔ | 3 warnings ✖ | 2 notes ✖
```

**Comment by mbojan** [comment]
Ready.

File: R/misc.R [hunk 1]
```diff
@@ -185,13 +194,127 @@ mixingmatrix.network <- function(object, attrname, expand.bipartite=FALSE, ...)
...
+}
+
+
+
+
+
+
+
+
+
```
Comment (krivit) [inline R/misc.R:317]: Can you please remove the extraneous whitespace?

File: R/misc.R [hunk 2]
```diff
@@ -369,36 +492,9 @@ is.discrete<-function(x){
...
-    rn <- c(rn, "Total")
-    cn <- c(cn, "Total")
-    if (x$type == "bipartite")
-      dimnames(m) <- list(B1 = rn,B2 = cn)
-    else
-      dimnames(m) <- list(From = rn,To = cn)
-  }
-  print(m)
-}
+
```
Comment (krivit) [inline R/misc.R:497]: Ditto.

**Review by krivit — CHANGES_REQUESTED** [review]
Thanks for the enhancement!

File: R/misc.R (hunk 2 above)
Comment (mbojan) [reply R/misc.R:497]: Will do

**Comment by krivit** [comment]
No comments from Carter, so I'll go ahead and merge.

**Pull request merged**